client/packages/invoices/src/Prescriptions/DetailView/columns.ts

```diff
@@ -305,19 +364,33 @@ export const usePrescriptionColumn = ({
       accessor: ({ rowData }) => {
         if ('lines' in rowData) {
           // Multiple lines, so we need to calculate the average price per unit
-
+          const { lines } = rowData;
+          const displayVaccineInDoses =
+            newPrefs?.displayVaccineInDoses && lines[0]?.item.isVaccine;
           let totalSellPrice = 0;
           let totalUnits = 0;
 
-          for (const line of rowData.lines) {
+          for (const line of lines) {
+            const units = line.numberOfPacks * line.packSize;
             totalSellPrice += line.sellPricePerPack * line.numberOfPacks;
-            totalUnits += line.numberOfPacks * line.packSize;
+
+            if (displayVaccineInDoses) {
+              totalUnits += units * line.item.doses;
```

Comment:
Same deal here, I think would be easier to reason about a new DosePrice column, than extra stuff here, as far as code is concerned, dose is a sub-unit, shouldn't be munged in with unit